plottable.d.ts

```diff
@@ -3056,6 +3056,32 @@ declare module Plottable {
              * @constructor
              */
             constructor();
+            /**
+             * Gets the interpolation function associated with the plot.
+             *
+             * @return {string | (points: Array<[number, number]>) => string)}
+             */
+            interpolator(): string | ((points: Array<[number, number]>) => string);
+            interpolator(interpolator: "linear"): Line<X>;
+            interpolator(interpolator: "linear-closed"): Line<X>;
+            interpolator(interpolator: "step"): Line<X>;
+            interpolator(interpolator: "step-before"): Line<X>;
+            interpolator(interpolator: "step-after"): Line<X>;
+            interpolator(interpolator: "basis"): Line<X>;
+            interpolator(interpolator: "basis-open"): Line<X>;
+            interpolator(interpolator: "basis-closed"): Line<X>;
+            interpolator(interpolator: "bundle"): Line<X>;
+            interpolator(interpolator: "cardinal"): Line<X>;
+            interpolator(interpolator: "cardinal-open"): Line<X>;
+            interpolator(interpolator: "cardinal-closed"): Line<X>;
+            interpolator(interpolator: "monotone"): Line<X>;
+            /**
+             * Sets the interpolation function associated with the plot.
+             *
+             * @param {string | points: Array<[number, number]>) => string} interpolator Interpolation function
+             * @return Plots.Line
+             */
+            interpolator(interpolator: string | ((points: Array<[number, number]>) => string)): Plots.Line<X>;
```

Comment:
This will not work because of this.  https://github.com/Microsoft/TypeScript/issues/1805

Also, current implementation mimics d3.d.ts type signature as can be seen here https://github.com/palantir/plottable/blob/757fix/typings/d3/d3.d.ts#L2223

More no this in a conversation here 
https://github.com/borisyankov/DefinitelyTyped/pull/5095